fem/coefficient.cpp

```diff
@@ -212,15 +212,16 @@ void GradientGridFunctionCoefficient::Eval(
 CurlGridFunctionCoefficient::CurlGridFunctionCoefficient (
    const GridFunction *gf)
    : VectorCoefficient ((gf) ?
-                        gf -> FESpace() -> GetMesh() -> SpaceDimension() : 0)
+                        (2 * gf -> FESpace() -> GetMesh() -> SpaceDimension()
```

Comment:
So just to confirm if I understand this correctly... in case of a 2D problem, a VectorCoefficient of dim=1 would be used instead of a Coefficient...is that right?

Reply:
That is correct.

Reply:
What will happen in 1D? The vector dimension will be negative, right? Maybe add an assert or use the logic `dim == 3 ? 3 : 1`, which is applied elsewhere and the object will be valid and it will just die at the evaluation.

Reply:
@najlkin , that's a good point.  The curl will always be zero in 1D and it's only defined in 2D or 3D.  I added an `MFEM_VERIFY` to check the space dimension and issue an error if the dimension is not 2 or 3.